contracts/escrow/dapps/AaveClaimAaveFacet.sol

```diff
@@ -0,0 +1,102 @@
+import { IStakedAave } from "../../shared/interfaces/IStakedAave.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract AaveClaimAaveFacet is PausableMods, DappMods {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @dev The address of Aave's Incentives controller Address on the deployed network
+     * @dev example - Aave's Incentives controller contract address on L1 mainnet or L2 polygon mainnet
+     */
+    address public immutable INCENTIVES_CONTROLLER_ADDRESS;
+    address public immutable STAKE_TOKEN_ADDRESS;
+
+    /**
+     * @notice Sets the network relevant address for Aave's Incentives controller Address on protocol deployment.
+     * @param aaveIncentivesControllerAddress The immutable address of Aave's Incentives controller Address on the deployed network.
+     */
+    constructor(
+        address aaveIncentivesControllerAddress,
+        address aaveStakeTokenAddress
+    ) public {
+        INCENTIVES_CONTROLLER_ADDRESS = aaveIncentivesControllerAddress;
+        STAKE_TOKEN_ADDRESS = aaveStakeTokenAddress;
+    }
+
+    /**
+        @notice This event is emitted every time Aave deposit is invoked successfully.
+        @param borrower address of the loan borrower.
+        @param loanID ID of the loan.
+     */
+    event AaveClaimed(address borrower, uint256 loanID);
+
+    /**
+     * @notice To claim AAVE call the {claimRewards} function on {AaveIncentivesController}.
+     * @param loanID id of the loan being used in the dapp
+     * @param amount amount of tokens to claim.
+     */
+    function aaveClaimAave(
+        uint256 loanID,
+        uint256 amount,
+        address[] calldata tokenAddresses
+    ) public paused("", false) onlyBorrower(loanID) {
+        address escrow = address(LibEscrow.e(loanID));
+        bytes memory result = LibEscrow.e(loanID).callDapp(
+            address(INCENTIVES_CONTROLLER_ADDRESS),
+            abi.encodeWithSelector(
+                IAaveIncentivesController.claimRewards.selector,
+                tokenAddresses,
+                amount,
+                escrow
+            )
+        );
+
+        bytes memory unstake = LibEscrow.e(loanID).callDapp(
+            address(STAKE_TOKEN_ADDRESS),
+            abi.encodeWithSelector(
+                IStakedAave.redeem.selector,
+                msg.sender,
+                amount
+            )
+        );
+
+        // Add AAVE to escrow token list
+        // TODO: get the AAVE token (add as immutable constructor variable)
+        // LibEscrow.tokenUpdated(loanID, address(0));
+
+        emit AaveClaimed(msg.sender, loanID);
+    }
+
+    /**
+     * @notice This function calculates the amount of aave tokens that can be redeemed.
+     * @param loanID id of the loan being used in the dapp
+     */
+    function aaveCalculateAave(uint256 loanID)
+        public
+        view
+        paused("", false)
+        onlyBorrower(loanID)
```

Comment:
is onlyBorrower necessary for a View method ?

Reply:
fixing these on `feature/claimAaveCompToUser`